src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -219,19 +220,44 @@ public Visitable visit(MethodDeclaration method, Void p) {
   @Override
   public Visitable visit(Parameter para, Void p) {
     if (insideTargetMethod) {
-      ResolvedType paraType = para.resolve().getType();
-      if (paraType.isReferenceType()) {
-        String paraTypeFullName =
-            paraType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
-        usedClass.add(paraTypeFullName);
-        for (ResolvedType typeParameterValue : paraType.asReferenceType().typeParametersValues()) {
-          String typeParameterValueName = typeParameterValue.describe();
-          if (typeParameterValueName.contains("<")) {
-            // removing the "<...>" part if there is any.
-            typeParameterValueName =
-                typeParameterValueName.substring(0, typeParameterValueName.indexOf("<"));
+      Type type = para.getType();
+      String paraTypeFullName = "";
+      ResolvedType paramType;
+      if (type.isUnionType()) {
+        for (var param : para.getType().asUnionType().getElements()) {
```

Comment:
Personally, I'm not a fan of `var`, since it is sometimes important to know the type of a variable. I'd prefer that we avoid it, even in new code.

Reply:
It's not any easier to write (it still takes keystrokes!), but it makes the code harder to read by obscuring information. So, it's a poor tradeoff for code that will need to be maintained (= all code we write for research!).

Reply:
Professor, I have used the actual type instead of `var`